app/helpers/templates_helper.rb

```diff
@@ -81,4 +86,15 @@ def template_input_f(f, options = {})
       react_form_input(input_type, f, :value, options)
     end
   end
+
+  def template_input_value_type_options
+    [[_('Plain'), 'plain'], [_("Search"), 'search'], [_('Date'), 'date'], [_('Resource'), 'resource']]
+  end
+
+  def resource_value_options(resource_type)
+    resource_type.constantize
+                 .authorized("view_#{resource_type.underscore.pluralize}".to_sym)
+                 .sort
+                 .map { |r| [r, r.id] }
```

Comment:
` .map { |r| [r, r.id] }` -> ` .map { |r| [r.to_s, r.id] }` would be slightly better, otherwise we just rely on ERB to convert the resource to string